raft-tech/TANF-app
# 1845 - Email on data submit

## Summary of Changes
Pull request closes #1845 

Sends an email to all `Data Analyst` users when a data file is uploaded for their location.

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && npm run start
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
1. In the django admin, give the group "Data Analyst" to your user (either the same or a different user that you will log into the frontend and upload a data file with) and save. Then, set the user's account approval status  to "Approved". Finally, set the user's location type to "STT" and give them a location ID.
1. Back to the frontend - navigate to the "Data Files" tab and select a fiscal year and quarter, upload a file.
1. An email should send to the "Data Analyst" user you set up for that location.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] A data submitted email is sent to all relevant users associated with the STT when data successfully submitted
+ [x] data submitted email is not sent out for unsuccessful data submissions or for any other activities on data-files page

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested? **N/A**
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
